master/buildbot/reporters/words.py

```diff
@@ -1257,3 +1254,60 @@ def clientConnectionLost(self, connector, reason):
 
     def clientConnectionFailed(self, connector, reason):
         reactor.callLater(self.failedDelay, connector.connect)
+
+
+class WebhookBotMixin(resource.Resource):
+    """
+    This is a mixin to be used by chat bots based on web-hooks
+    """
+
+    def __init__(self, path):
+        resource.Resource.__init__(self)
+        www = get_plugins('www', None, load_now=True)
+        if 'base' not in www:
+            raise RuntimeError("could not find buildbot-www; is it installed?")
+        self._root = www.get('base').resource
+        self.path = path
+
+    def startService(self):
```

Comment:
I think this will be cleaner to implement that as a child service. Mixins are supposed to not override any method of the parent class (and not have an __init__) to avoid super() ordering hell.
Mixin can only inherit from object(). You can create a WebhookBotMixin, but he has to maintain its resource as a child service.

If we want to refactor this for reuse, we need to make it right.